CMakeLists.txt

```diff
@@ -593,9 +596,11 @@ set(CPACK_PACKAGE_VENDOR "Apache NiFi")
 set(CPACK_PACKAGE_VERSION_MAJOR ${PROJECT_VERSION_MAJOR})
 set(CPACK_PACKAGE_VERSION_MINOR ${PROJECT_VERSION_MINOR})
 set(CPACK_PACKAGE_VERSION_PATCH ${PROJECT_VERSION_PATCH})
-set(CPACK_PACKAGE_DESCRIPTION_FILE "${CMAKE_CURRENT_SOURCE_DIR}/README.md")
-configure_file("${CMAKE_CURRENT_SOURCE_DIR}/LICENSE" "${CMAKE_CURRENT_BINARY_DIR}/LICENSE.txt" COPYONLY)
-set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_BINARY_DIR}/LICENSE.txt")
+if(NOT MINIFI_BUILD_CONAN_PACKAGE)
+    set(CPACK_PACKAGE_DESCRIPTION_FILE "${CMAKE_CURRENT_SOURCE_DIR}/README.md")
+    configure_file("${CMAKE_CURRENT_SOURCE_DIR}/LICENSE" "${CMAKE_CURRENT_BINARY_DIR}/LICENSE.txt" COPYONLY)
+    set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_BINARY_DIR}/LICENSE.txt")
+endif()
```

Comment:
Why are these disabled in conan packages? We need to include the LICENSE and NOTICE files in every distribution, including binary ones.

Reply:
When we create the **MiNiFi C++ conan package**, **LICENSE** and **NOTICE** are in the conan package. In the **conanfile.py**, we tell conan to include all the files assigned to **shared_sources** variable to be included in the conan package. In our conanfile.py's **package()** method, we also run **cmake.install()**, so in MiNiFi C++'s CMakeLists.txt files all CMake commands using **install(...)** will also copy over those sources (.h*, .i*, etc) and compiled binary assets (static libs, shared libs, minifi binary, etc) into our MiNiFi C++ conan package.

In the past when I tried maintaining that CPACK part way of doing it, when I ran **conan create**, it complained about those 3 lines. However, today with **conan 2.0.17** I removed that MINIFI_BUILD_CONAN_PACKAGE check and was able to successfully run **conan create** with those 3 lines enabled. I will push the update commit up to the repo now.

Reply:
Heres a view of my minifi cpp conan package p/ folder where you can see **LICENSE, NOTICE** and other assets copied over.

![minifi_cpp_conan_package_folder_view_aug92024](https://github.com/user-attachments/assets/dffffa60-04b9-4ca4-b28a-f97dfec4137e)